src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -969,6 +969,11 @@ OverlayImpl::send (protocol::TMValidation& m)
         if (! m.has_hops() || p->hopsAware())
             p->send(sm);
     });
+    STValidation::pointer val;
+    SerialIter sit (makeSlice(m.validation()));
+    val = std::make_shared <
+        STValidation> (std::ref (sit), false);
+    app_.getOPs().pubValidation (val);
```

Comment:
It's not ideal to deserialize the validation we just serialized, but the alternative is to call `pubValidation` everywhere we create a validation.